Imports

In [1]:
import pandas as pd
import numpy as np

Load dataset

In [2]:
df = pd.read_csv("synthetic_coffee_health_10000.csv")

Properties of the dataset

In [ ]:
print("Shape:", df.shape)    # rows x cols

Shape: (10000, 16)


In [5]:
print("Column types:")
print(df.dtypes)

Column types:
ID                           int64
Age                          int64
Gender                      object
Country                     object
Coffee_Intake              float64
Caffeine_mg                float64
Sleep_Hours                float64
Sleep_Quality               object
BMI                        float64
Heart_Rate                   int64
Stress_Level                object
Physical_Activity_Hours    float64
Health_Issues               object
Occupation                  object
Smoking                      int64
Alcohol_Consumption          int64
dtype: object


In [7]:
print("First 5 rows:")
display(df.head())

First 5 rows:


ID  Age  Gender  Country  Coffee_Intake  Caffeine_mg  Sleep_Hours  \
0   1   40    Male  Germany            3.5        328.1          7.5   
1   2   33    Male  Germany            1.0         94.1          6.2   
2   3   42    Male   Brazil            5.3        503.7          5.9   
3   4   53    Male  Germany            2.6        249.2          7.3   
4   5   32  Female    Spain            3.1        298.0          5.3   

  Sleep_Quality   BMI  Heart_Rate Stress_Level  Physical_Activity_Hours  \
0          Good  24.9          78          Low                     14.5   
1          Good  20.0          67          Low                     11.0   
2          Fair  22.7          59       Medium                     11.2   
3          Good  24.7          71          Low                      6.6   
4          Fair  24.1          76       Medium                      8.5   

  Health_Issues Occupation  Smoking  Alcohol_Consumption  
0           NaN      Other        0                    0  
1           NaN    Service        0                    0  
2          Mild     Office        0                    0  
3          Mild      Other        0                    0  
4          Mild    Student        0                    1

Check for missing values

In [8]:
print("Missing values:")
print(df.isnull().sum())

Missing values:
ID                            0
Age                           0
Gender                        0
Country                       0
Coffee_Intake                 0
Caffeine_mg                   0
Sleep_Hours                   0
Sleep_Quality                 0
BMI                           0
Heart_Rate                    0
Stress_Level                  0
Physical_Activity_Hours       0
Health_Issues              5941
Occupation                    0
Smoking                       0
Alcohol_Consumption           0
dtype: int64


The Health_Issues column has 5941 missing values out of 10000 entries. We will inspect this further.

In [10]:
print(df["Health_Issues"].value_counts())

Health_Issues
Mild        3579
Moderate     463
Severe        17
Name: count, dtype: int64


Looking at the dataset, the 5941 missing values in the Health_Issues column are actually just None - as in no health issues. The system interprets this to be the None keyword = missing value. So we will rename all of these entries so they are no longer considered as missing values.

In [13]:
df["Health_Issues"] = df["Health_Issues"].fillna("No_issues")
print(df["Health_Issues"].value_counts())

Health_Issues
No_issues    5941
Mild         3579
Moderate      463
Severe         17
Name: count, dtype: int64


Now, we check again for missing values.

In [14]:
print("Missing values:")
print(df.isnull().sum())

Missing values:
ID                         0
Age                        0
Gender                     0
Country                    0
Coffee_Intake              0
Caffeine_mg                0
Sleep_Hours                0
Sleep_Quality              0
BMI                        0
Heart_Rate                 0
Stress_Level               0
Physical_Activity_Hours    0
Health_Issues              0
Occupation                 0
Smoking                    0
Alcohol_Consumption        0
dtype: int64


There are no missing values now.